### PR TITLE
Log all COR_, CORECLR_, DOTNET_, and OTEL_environmental variables in native code

### LINKED ISSUE #1325: Improve env vars internal logging
It would be better to log all env vars having the following prefixes:
- `COR_`
- `CORECLR_`
- `DOTNET_`
- `OTEL_`

_Originally posted by @pellared in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1324#discussion_r985612360_

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Fixes #1325

## What

Log all COR_, CORECLR_, DOTNET_, and OTEL_environmental variables in native code

## Tests

Manual testing with different set of variables.

## Checklist

- ~~[ ] `CHANGELOG.md` is updated.~~
- ~~[ ] Documentation is updated.~~
- ~~[ ] New features are covered by tests.~~

### CODE REVIEW

**Review by dszmigielski — APPROVED** [review]

File: src/OpenTelemetry.AutoInstrumentation.Native/util.cpp [hunk 1]
```diff
@@ -97,6 +99,53 @@ std::vector<WSTRING> GetEnvironmentValues(const WSTRING& name)
...
 
+
+std::vector<WSTRING> GetEnvironmentVariables(const std::vector<WSTRING> &prefixes)
+{
+    std::vector<WSTRING> env_strings;
+#ifdef _WIN32
+    auto env_variables = GetEnvironmentStrings();
+    int prev = 0;
+    for (int i = 0;; i++) {
+        if (env_variables[i] == '\0')
```
Comment (RassK) [inline src/OpenTelemetry.AutoInstrumentation.Native/util.cpp:110]: nit: I would revert this to reduce nesting.

**Review by RassK — APPROVED** [review]
single nit maybe worth considering.

File: src/OpenTelemetry.AutoInstrumentation.Native/util.cpp [hunk 2]
```diff
@@ -97,6 +99,53 @@ std::vector<WSTRING> GetEnvironmentValues(const WSTRING& name)
     return GetEnvironmentValues(name, L',');
 }
 
+
+std::vector<WSTRING> GetEnvironmentVariables(const std::vector<WSTRING> &prefixes)
+{
+    std::vector<WSTRING> env_strings;
+#ifdef _WIN32
+    auto env_variables = GetEnvironmentStrings();
```
Comment (pjanotti) [inline src/OpenTelemetry.AutoInstrumentation.Native/util.cpp:107]: The remark linked below seems very useful to understand what is going on here: https://learn.microsoft.com/en-us/windows/win32/api/processenv/nf-processenv-getenvironmentstrings#remarks

Comment (pjanotti) [reply src/OpenTelemetry.AutoInstrumentation.Native/util.cpp:107]: Like done for non-Windows this should also be const.

File: src/OpenTelemetry.AutoInstrumentation.Native/util.cpp [hunk 3]
```diff
@@ -97,6 +99,53 @@ std::vector<WSTRING> GetEnvironmentValues(const WSTRING& name)
...
+            if (env_variables[i + 1] == '\0')
+            {
+                break;
+            }
+        }
+    }
+
+    FreeEnvironmentStrings(env_variables);
+#else
+    const auto env_variables = environ;
```
Comment (pjanotti) [inline src/OpenTelemetry.AutoInstrumentation.Native/util.cpp:132]: I don't see this one used anywhere.

File: src/OpenTelemetry.AutoInstrumentation.Native/util.cpp [hunk 4]
```diff
@@ -97,6 +99,53 @@ std::vector<WSTRING> GetEnvironmentValues(const WSTRING& name)
...
+    for (char** current = environ; *current; current++)
```
Comment (pjanotti) [inline src/OpenTelemetry.AutoInstrumentation.Native/util.cpp:133]: If removing the unused above then add the proper const to `current` here.

Comment (Kielek) [reply src/OpenTelemetry.AutoInstrumentation.Native/util.cpp:133]: It is not possible as current is incremented in last part of for loop: `current++`

File: src/OpenTelemetry.AutoInstrumentation.Native/util.cpp (hunk 1 above)
Comment (Kielek) [reply src/OpenTelemetry.AutoInstrumentation.Native/util.cpp]: Fixed: 9f1d8300e51eda9af97d7eb928e44a5f815a3bd5

File: src/OpenTelemetry.AutoInstrumentation.Native/util.cpp (hunk 2 above)
Comment (Kielek) [reply src/OpenTelemetry.AutoInstrumentation.Native/util.cpp]: Both fixed: 93f0408db630f7682e1f27a9235a2dc85cb83a0d

File: src/OpenTelemetry.AutoInstrumentation.Native/util.cpp (hunk 3 above)
Comment (Kielek) [reply src/OpenTelemetry.AutoInstrumentation.Native/util.cpp]: Fixed: 3b381379e5b431e2cad3854feefae7228b6fd465

**Review by pellared — APPROVED** [review]

**Pull request merged**